Tests/CombineWampTests/MessageJSONTests.swift

```diff
@@ -0,0 +1,2577 @@
+        // given
+        let decoder = JSONDecoder()
+        let json = """
+                   [50, 7814135, {}, [30]]
+                   """.data(using: .utf8)!
+        let model = Message.result(
+            .init(
+                request: .init(integerLiteral: 7814135),
+                details: [:],
+                arguments: [.integer(30)],
+                argumentsKw: nil
+            )
+        )
+
+        // when
+        let message = try decoder.decode(Message.self, from: json)
+
+        // then
+        XCTAssertEqual(model, message)
+    }
+
+    // https://wamp-proto.org/_static/gen/wamp_latest.html#result-0
+    func testResultExample3Encoding() throws {
+        // given
+        let encoder = JSONEncoder()
+        encoder.outputFormatting = .sortedKeys
+        let json = """
+                   [50,7814135,{},[30]]
+                   """
+        let model = Message.result(
+            .init(
+                request: .init(integerLiteral: 7814135),
+                details: [:],
+                arguments: [.integer(30)],
+                argumentsKw: nil
+            )
+        )
+
+        // when
+        let message = try encoder.encode(model)
+
+        // then
+        XCTAssertEqual(json, String(data: message, encoding: .utf8)!)
+    }
+
+    // https://wamp-proto.org/_static/gen/wamp_latest.html#result-0
+    func testResultExample4Decoding() throws {
+        // given
+        let decoder = JSONDecoder()
+        let json = """
+                   [50, 7814135, {}, [], {"userid": 123, "karma": 10}]
+                   """.data(using: .utf8)!
+        let model = Message.result(
+            .init(
+                request: .init(integerLiteral: 7814135),
+                details: [:],
+                arguments: [],
+                argumentsKw: [
+                    "userid": .integer(123),
+                    "karma": .integer(10)
+                ]
+            )
+        )
+
+        // when
+        let message = try decoder.decode(Message.self, from: json)
+
+        // then
+        XCTAssertEqual(model, message)
+    }
+
+    // https://wamp-proto.org/_static/gen/wamp_latest.html#result-0
+    func testResultExample4Encoding() throws {
+        // given
+        let encoder = JSONEncoder()
+        encoder.outputFormatting = .sortedKeys
+        let json = """
+                   [50,7814135,{},[],{"karma":10,"userid":123}]
+                   """
+        let model = Message.result(
+            .init(
+                request: .init(integerLiteral: 7814135),
+                details: [:],
+                arguments: [],
+                argumentsKw: [
+                    "userid": .integer(123),
+                    "karma": .integer(10)
+                ]
+            )
+        )
+
+        // when
+        let message = try encoder.encode(model)
+
+        // then
+        XCTAssertEqual(json, String(data: message, encoding: .utf8)!)
+    }
+}
```

Comment:
I really really like the test coverage :)

Reply:
Yeah, a bit large file, but I guess this is necessary for JSONs.